distributed/utils_test.py

```diff
@@ -1567,3 +1569,32 @@ def transition(self, key, start, finish, **kwargs):
             ts.metadata["start_time"] = time()
         elif start == "executing" and finish == "memory":
             ts.metadata["stop_time"] = time()
+
+
+def set_config_and_reload(cfg):
+    """Test function decorator that temporarily sets dask.config variables and updates
+    the cached copy in the global variables of the distributed.scheduler module.
+
+    Tests decorated with this function are skipped when Cython is enabled.
+
+    Note: this decorator must be placed *before* @gen_cluster.
+    """
+    from . import scheduler
+
+    def decorator(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            try:
+                with dask.config.set(cfg):
+                    importlib.reload(scheduler)
+                    return f(*args, **kwargs)
+            finally:
+                importlib.reload(scheduler)
+
+        if scheduler.COMPILED:
+            return pytest.mark.skip(
+                "Monkey-patching module variables doesn't work with Cython"
+            )(wrapper)
+        return wrapper
+
+    return decorator
```

Comment:
@jakirkham I could not make this work with Cython; could you take a look at it?
scheduler.py caches a bunch of dask.config keys for speed as global variables; I could not find a way to tamper with them when the module is cythonized.

- the design above does nothing with cython, in other words importlib.reload() seems to have no effect
- I also tried to monkey-patch the global variables themselves, but I get an AttributeError when Cython is enabled. I already tried changing scheduler.py to ``declare(..., visibility="public")`` but it does not seem to have an effect.

I guess a possible solution would be to move the dask.config.get calls from the module definition to ``Scheduler.__init__``?